packages/apps/patterns/react-appkit/src/components/CrudList/CrudList.tsx

```diff
@@ -0,0 +1,226 @@
+          }
+          break;
+        }
+        case 'PageUp': {
+          if (targetIndex > 0) {
+            event.preventDefault();
+            inputsInScope[targetIndex - 1].focus();
+          }
+          break;
+        }
+      }
+    },
+    [hostId]
+  );
+  return { onListItemInputKeyDown, hostAttrs, itemAttrs };
+};
+
+export const CrudList = ({
+  children,
+  labelId,
+  nextItemTitle,
+  defaultNextItemTitle,
+  onNextItemTitleChange,
+  defaultItemIdOrder,
+  itemIdOrder,
+  onItemIdOrderChange,
+  slots = {}
+}: CrudListProps) => {
+  const [listItemIds, setListItemIds] = useControllableState({
+    prop: itemIdOrder,
+    defaultProp: defaultItemIdOrder,
+    onChange: onItemIdOrderChange
+  });
+  const handleDragEnd = (event: DragEndEvent) => {
+    const { active, over } = event;
+    if (active.id !== over?.id) {
+      setListItemIds((itemIds) => {
+        if (itemIds && over?.id) {
+          const oldIndex = itemIds.findIndex((id) => id === active.id);
+          const newIndex = itemIds.findIndex((id) => id === over.id);
+          return arrayMove(itemIds, oldIndex, newIndex);
+        } else {
+          return itemIds;
+        }
+      });
+    }
+  };
+  return (
+    <div role='none' {...slots.root} className={mx('contents', slots.root?.className)}>
+      <List
+        // ref={forwardedRef}
+        labelId={labelId}
+        variant='ordered-draggable'
+        selectable
+        onDragEnd={handleDragEnd}
+        listItemIds={listItemIds ?? []}
+        slots={slots.list}
+      >
+        {children}
+      </List>
+      <div className='flex'>
+        <ListItemDragHandle className='invisible' />
+        <ListItemEndcap className='invisible' />
+        <Input
+          variant='subdued'
+          label='Add new item'
+          placeholder='Add new item'
+          {...{
+            value: nextItemTitle,
+            defaultValue: defaultNextItemTitle,
+            onChange: onNextItemTitleChange
+          }}
+          slots={{
+            input: {
+              ...slots.addItem?.input,
+              className: mx('p-1 mbs-1', slots.addItem?.input?.className)
+            },
+            root: {
+              ...slots.addItem?.root,
+              className: mx('grow mbs-0', slots.addItem?.root?.className)
+            },
+            label: {
+              ...slots.addItem?.label,
+              className: mx('sr-only', slots.addItem?.label?.className)
+            }
+          }}
+        />
+        <ListItemEndcap className='invisible' />
+      </div>
+    </div>
+  );
+};
+
+export const CrudListItem = forwardRef<HTMLLIElement, CrudListItemProps>(
+  (
+    {
+      id,
+      defaultCompleted,
+      completed,
+      onCompletedChange,
+      defaultTitle,
+      title,
+      onTitleChange,
+      slots = {}
+    }: CrudListItemProps,
+    forwardedRef
+  ) => {
+    return (
+      <ListItem
+        ref={forwardedRef}
+        {...{
```

Comment:
why `...`?

Reply:
I frequently use object spreads instead of the equivalent `propName={propValue}` to take advantage of the same-key-name-as-variable-name shorthands, though I treat both syntaxes as equivalent for values that are not `key` (which I prefer to always use the usual JSX syntax because `key` is special).

I don't feel strongly about this pattern. If it’s causing ergonomic issues, let’s find a formatter rule that we think works best.